erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -134,17 +127,17 @@ def validate_against_jv(self):
 					frappe.throw(_("For {0}, only debit accounts can be linked against another credit entry")
 						.format(d.account))
 
-				if d.against_jv == self.name:
+				if d.reference_name == self.name:
 					frappe.throw(_("You can not enter current voucher in 'Against Journal Entry' column"))
 
 				against_entries = frappe.db.sql("""select * from `tabJournal Entry Account`
 					where account = %s and docstatus = 1 and parent = %s
-					and ifnull(against_jv, '') = '' and ifnull(against_invoice, '') = ''
-					and ifnull(against_voucher, '') = ''""", (d.account, d.against_jv), as_dict=True)
+					and ifnull(reference_type, '') = '' and ifnull(reference_name, '') = ''
```

Comment:
reference_type in ('', 'Sales Order', 'Purchase Order')
remove reference_name condition